### PR TITLE
[Hexagon] Account for objects being smaller than the allocated space

### PR DESCRIPTION
**Pull request by kparzysz-quic**
In particular, graph executor will reuse allocated buffers for various tensors. These tensors may be of various sizes as long as the buffer is large enough to hold them.

### CODE REVIEW

**Comment by tmoreau89** [comment]
CC @csullivan @Lunderberg @adstraw

File: src/runtime/hexagon/hexagon/hexagon_buffer.cc
Comment (adstraw) [inline src/runtime/hexagon/hexagon/hexagon_buffer.cc:234]: Nit:  It's called `copied` here and `offset` in `CopyTo` and `CopyFrom` above.  The concept is the exact same, I believe, but the variable names are different which could cause some confusion.  I don't have a preference on which name so long as it's consistent.

File: src/runtime/hexagon/hexagon/hexagon_buffer.cc
Comment (adstraw) [inline src/runtime/hexagon/hexagon/hexagon_buffer.cc:236]: Good catch and change!

**Review by adstraw — APPROVED** [review]
Looks *very* good to me.  Thanks for this contribution @kparzysz-quic.

File: src/runtime/hexagon/hexagon/hexagon_buffer.cc
Comment (kparzysz-quic) [reply src/runtime/hexagon/hexagon/hexagon_buffer.cc:234]: Yeah, I didn't want to modify the existing code too much, so I left `offset` unchanged.  I'll merge it to unblock our tests, but I'll create another PR to make the names consistent.  Thanks Adam!

Comment (kparzysz-quic) [reply src/runtime/hexagon/hexagon/hexagon_buffer.cc:234]: Apparently this PR still needs approval from someone else, so I just added another commit that renames `offset` to `copied`.

**Review by junrushao1994 — APPROVED** [review]

**Comment by adstraw** [comment]
@kparzysz-quic you should be good to merge now; thanks @junrushao1994 for reviewing!

**Comment by junrushao1994** [comment]
Thanks @kparzysz-quic @adstraw! Please enjoy the holiday BTW!

**Pull request merged**
